### PR TITLE
Create offline storage context wrapper

### PR DESCRIPTION
**Pull request by melsoriano**
Created IndexedDB offline storage context wrapper

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -26,12 +27,24 @@ async function enableMocking() {
   });
 }
 
+const offlineStorageConfig = {
+  type: "indexedDB",
+  name: import.meta.env.VITE_INDEXED_DB_NAME || "radfish_dev",
```
Comment (brockinit) [inline src/index.jsx:32]: Recalling what Jesse had mentioned, I think we should avoid setting a default db name and instead let it error out if they don't supply one.

Comment (brockinit) [reply src/index.jsx:32]: Along with provide a warning as to what's missing

File: src/packages/contexts/OfflineStorageWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <OfflineStorageContext.Provider value={contextValue}>{children}</OfflineStorageContext.Provider>
+  );
+};
+
+export const useOfflineStorage = () => {
+  const context = useContext(OfflineStorageContext);
+  if (!context) {
+    throw new Error("useOfflineStorage must be used within an OfflineStorageWrapper");
```
Comment (brockinit) [inline src/packages/contexts/OfflineStorageWrapper.jsx:45]: It seems like during the build `context` is missing, causing it to hit line 45?

File: src/index.jsx [hunk 3]
```diff
@@ -26,12 +27,24 @@ async function enableMocking() {
   });
 }
 
+const offlineStorageConfig = {
```
Comment (brockinit) [inline src/index.jsx:30]: Please provide a link to documentation here

**Review by brockinit — CHANGES_REQUESTED** [review]
Left some comments! I also didn't see data show up in the `formData` table when I submitted the ComplexForm 
image

**Review by brockinit — COMMENTED** [review]
Had one other question:

- What patterns can we follow to tie `useOfflineStorage` into the `FormWrapper` and `TableWrapper`? We want to ensure that things like onChange events are storing data into IndexDB as well as the React state. It doesn't look like `useOfflineStorage` is being used in the wrappers. It's just an unused import

File: src/contexts/FormWrapper.example.jsx [hunk 4]
```diff
@@ -91,6 +91,19 @@ export const FormWrapper = ({ children, onSubmit }) => {
     handleComputedValuesLogic(inputIds, formData, FORM_CONFIG);
   }, []);
 
+  const saveOfflineData = async (tableName, data) => {
+    try {
+      if (params.id) {
+        await updateOfflineData(tableName, [{ uuid: params.id, ...data }]);
+      }
+      // else {
```
Comment (brockinit) [inline src/contexts/FormWrapper.example.jsx:99]: Does this need to be commented out? @melsoriano

**Review by brockinit — APPROVED** [review]
We need to address the other comments left above, but I'm merging this for now so that we can ship release 0.8.0

**Comment by gambinish** [comment]
Looks like tests are failing. I'm down to pair on this if you want.

**Pull request merged**